subprojects/base-services/src/main/java/org/gradle/internal/concurrent/ManagedExecutorImpl.java

```diff
@@ -83,7 +83,8 @@ public void stop(int timeoutValue, TimeUnit timeoutUnits) throws IllegalStateExc
                 throw new IllegalStateException("Timeout waiting for concurrent jobs to complete.");
             }
         } catch (InterruptedException e) {
-            throw new UncheckedException(e);
+            executor.shutdownNow();
+            throw UncheckedException.throwAsUncheckedException(e);
```

Comment:
I think this is not the right way to handle this. We know explicitly (as we should always know explicitly) that we are dealing with an `InterruptedException`, so there is no need for wrapping in `throwAsUncheckedException()`. We could have something like `Interrputs.rethrowAndInterruptThread(e)` for this specific purpose.

Reply:
Same reasoning as above - I first wanted to have a consistent idiom before we start the discussion around what to do with UncheckedException.

Reply:
I still fail to see how this is better. A consistent idiom for what exactly?

Reply:
A consistent idiom for using UncheckedException. Sometimes people used a constructor, sometimes they used the rethrow method. I don't see a reason to have both. Now we can easily search for all uses of the rethrow method and remove it where it is not appropriate.